src/core/events.js

```diff
@@ -40,8 +40,7 @@ export class Events {
    * @param {object} state - The state object { G, ctx }.
    */
   update(state) {
-    const length = this.dispatch.length;
-    for (let i = 0; i < length; i++) {
+    for (let i = 0; i < this.dispatch.length; i++) {
```

Comment:
Let's limit this to 100 iterations to catch infinite loops. We can probably log a message if this loop reaches the 100th iteration.

Reply:
It's possible that some other code in `flow.js` is now catching infinite loops.

Reply:
I’ve tried adding two more tests to see if I can trigger an infinite loop, but they seem to be caught elsewhere. Given these tests, maybe it’s safe to assume that an iteration limit is unnecessary?

Reply:
OK, sounds good to me.